### PR TITLE
Possibility to turn off the solvers and initial conditions setup during the initial refinement

### PR DESCRIPTION
**Pull request by LudovicJnnt**
For those who want to skip solvers and the initial conditions setup, this pull request adds 2 parameters for the subsection mesh refinement.

AssertThrow for solvers ON while setup initial conditions OFF does not work for some reason.
AssertThrow for skipping setup initial conditions while running postprocessors on initial refinement does not work either.

### CODE REVIEW

**Comment by jaustermann** [comment]
This addresses issue #1690. Looks like astyle test failed, @LudovicJnnt did you run astyle on the submitted code?

File: source/simulator/core.cc [hunk 1]
```diff
@@ -1591,6 +1591,16 @@ namespace aspect
 
     if (parameters.resume_computation == false)
       {
+        // See if we want to skip the initial conditions setup during the pre-
+        // refinement phase. If set true, the solvers won't be executed. An
```
Comment (gassmoeller) [inline source/simulator/core.cc]: If set true, the solvers won't be executed -> If set **to** true, the solver **will not** be executed/

**Comment by LudovicJnnt** [comment]
I added 2 tests for skipping solvers and/or initial conditions setup during the pre-refinement
I applied the changes from Rene.

**Comment by gassmoeller** [comment]
Let me see if I can simplfy the logic somewhat. After that this is ready to merge.

**Comment by gassmoeller** [comment]
@LudovicJnnt: I updated the PR with a different logic that should produce the same result. Can you update your local branch and check that your models still work (do a git pull from this PR). The tests still seem to work.

**Comment by LudovicJnnt** [comment]
Right, yes models still work. 
Thank you!

**Comment by gassmoeller** [comment]
Then this is ready to merge when the tester is happy. Also please add an entry into doc/modules/changes and add a few sentences into the hackathon report.

**Pull request merged**